### PR TITLE
Scrub references to WebRequest from examples

### LINKED ISSUE #7712: Scrub references to WebRequest
There are some places within docs that discuss `WebRequest` as if it is a current solution e.g. https://docs.microsoft.com/en-us/dotnet/api/system.net?view=net-6.0

We should search all docs and ensure we don't present `WebRequest` or `WebClient` as a recommended solution. Instead docs should just be factual about functionality and how to use the APIs.

### PR DESCRIPTION
**Pull request by rzikm**
## Summary

Replaced by references to HttpClient or client-agnostic references.

Fixes #7712

### CODE REVIEW

**Comment by rzikm** [comment]
All remaining snippets build failures are from C++/CLI

**Review by gewarren — APPROVED** [review]
This is super, thanks @rzikm!

File: snippets/cpp/VS_Snippets_Remoting/HttpVersion_Version10/CPP/httpversion_version10.cpp [hunk 1]
```diff
@@ -1,46 +1,49 @@
...
-The 'Version10' field of the 'HttpVersion' class is assigned to the 'ProtocolVersion' property of the 'HttpWebRequest' Class.
-The changed Version and the 'ProtocolVersion' of the response Object* are displayed.
+The 'Version'  property of 'HttpRequestMessage' class identifies the Version of HTTP being used.
+Then the default value of 'Version' property is displayed to the console.
+The 'Version10' field of the 'HttpVersion' class is assigned to the 'Version' property of the 'HttpRequestMessage' Class.
+The changed Version and the 'Version' of the response object are displayed.
 */
 
-#using <System.dll>
+#using < System.dll>
```
Comment (liveans) [inline snippets/cpp/VS_Snippets_Remoting/HttpVersion_Version10/CPP/httpversion_version10.cpp:9]: nit:
```suggestion
#using <System.dll>
```

**Review by liveans — APPROVED** [review]

File: snippets/csharp/System.Net/HttpVersion/Overview/httpversion_version10.cs [hunk 2]
```diff
@@ -18,28 +18,31 @@ public static void Main()
...
+			HttpClient client = new HttpClient();
+
+			HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, "http://www.microsoft.com");
+			Console.WriteLine("Default HTTP request version is {0}", request.Version);
+
+			request.Version = HttpVersion.Version10;
+			Console.WriteLine("Request version after assignment is {0}", request.Version);
+
+			HttpResponseMessage response = client.Send(request);
+			Console.WriteLine("Response HTTP version {0}", response.Version);
```
Comment (stephentoub) [inline snippets/csharp/System.Net/HttpVersion/Overview/httpversion_version10.cs:32]: The HttpClient instead and the HttpResponseMessage should both be disposed / should both be in using blocks / declarations.

File: xml/System.Net/WebProxy.xml [hunk 3]
```diff
@@ -80,7 +80,7 @@
    
   
 ## Examples  
- The following code example assigns a <xref:System.Net.WebProxy> instance to a <xref:System.Net.WebRequest> instance. The <xref:System.Net.WebRequest> instance uses the proxy to connect to external Internet resources. (For an example that demonstrates using the WPAD feature, see the documentation for the <xref:System.Net.IWebProxyScript> class.)  
+ The following code example sets up <xref:System.Net.Http.HttpClient> instance with a <xref:System.Net.WebProxy> instance. The <xref:System.Net.Http.HttpClient> instance uses the proxy to connect to external Internet resources. (For an example that demonstrates using the WPAD feature, see the documentation for the <xref:System.Net.IWebProxyScript> class.)
```
Comment (stephentoub) [inline xml/System.Net/WebProxy.xml:83]: > sets up HttpClient instance

"sets up _an_ HttpClient instance"

File: xml/System.Net/WebProxy.xml [hunk 4]
```diff
@@ -1145,7 +1145,7 @@
```
Comment (stephentoub) [inline xml/System.Net/WebProxy.xml:1145]: GetDefaultProxy throws PlatformNotSupportedException on .NET Core.

**Pull request merged**